editor/inspector_dock.cpp

```diff
@@ -399,6 +399,23 @@ void InspectorDock::_resource_selected(const Ref<Resource> &p_res, const String
 	EditorNode::get_singleton()->push_item(r.operator->(), p_property);
 }
 
+void InspectorDock::_files_moved(const String &p_old_file, const String &p_new_file) {
+	// Because only the file name is shown, we care about changes on the file name.
+	if (p_old_file.get_file() != p_new_file.get_file()) {
+		ObjectID current_id = EditorNode::get_singleton()->get_editor_selection_history()->get_current();
+		Object *current_obj = current_id.is_valid() ? ObjectDB::get_instance(current_id) : nullptr;
+		if (current_obj == nullptr) {
+			return;
+		}
+		Resource *res = Object::cast_to<Resource>(current_obj);
+		// We only care about updating the path if the current object is the one being renamed.
+		if (res != nullptr && p_old_file == res->get_path()) {
+			res->set_path(p_new_file);
```

Comment:
The Resource's path gets auto-updated without that, this line shouldn't be necessary and I think it can break things.
I'm not sure what normally updates the path though.

Reply:
Sorry for the silly question, but, Where Resource's path is auto-updated?
My initial implementation was only calling `object_selector->update_path();` but that didn't get it the new path, so I added `set_path()` call.

Reply:
Just in case I was misremembering I removed the call to `set_path` and my fix stopped working.
I wasn't able to find where Resource's path is auto-updated, but I guess that happens sometime after `files_moved` signal is emitted so when `EditorObjectSelector` reads the Resource's path on `update_path` is not changed yet.

Reply:
The paths are normally updated here:
https://github.com/godotengine/godot/blob/594d64ec244b6b99321e2096987d4d69de4c8845/editor/filesystem_dock.cpp#L1567-L1570
(at least for some resource types)

I checked whether setting path earlier would interfere and it doesn't.